sentry/src/main/java/io/sentry/util/InitUtil.java

```diff
@@ -13,14 +14,23 @@ public static boolean shouldInit(
       final @Nullable SentryOptions previousOptions,
       final @NotNull SentryOptions newOptions,
       final boolean isEnabled) {
-    if (SentryIntegrationPackageStorage.getInstance()
-        .checkForMixedVersions(newOptions.getLogger())) {
+    if (Platform.isJvm() && newOptions.getVersionDetector() instanceof NoopVersionDetector) {
+      newOptions.setVersionDetector(new ManifestVersionDetector(newOptions));
+    }
+    if (newOptions.getVersionDetector().checkForMixedVersions()) {
       newOptions
           .getLogger()
           .log(
               SentryLevel.ERROR,
               "Not initializing Sentry because mixed SDK versions have been detected.");
-      return false;
+      final @NotNull String docsUrl =
+          Platform.isAndroid()
+              ? "https://docs.sentry.io/platforms/android/troubleshooting/mixed-versions"
+              : "https://docs.sentry.io/platforms/java/troubleshooting/mixed-versions";
+      throw new IllegalStateException(
+          "Sentry SDK has detected a mix of versions. This is not supported and likely leads to crashes. Please always use the same version of all Java SDK modules (dependencies). See "
```

Comment:
```suggestion
          "Sentry SDK has detected a mix of versions. This is not supported and likely leads to crashes. Please always use the same version of all SDK modules (dependencies). See "
```
maybe that's better to not confuse Android users